### PR TITLE
Cleanup ISOTPScan unit tests

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

**Comment by gpotter2** [comment]
```
###(008)=[failed] 20000 bytes send
>>> data = dhex("01 02 03 04 05")*4000
>>> msg = ISOTP(data, dst=0x641)
>>> ready = threading.Event()
>>> fragments = msg.fragment()
>>> ack = CAN(identifier=0x241, data=dhex("30 00 00"))
>>> 
>>> def sender():
...     ready.wait(timeout=5)
...     s.send(msg)
... 
>>> thread = threading.Thread(target=sender, name="sender")
>>> thread.start()
>>> 
>>> ready.set()
>>> ff = stim.sniff(count=1, timeout=1)[0]
>>> assert (bytes(ff) == bytes(fragments[0]))
>>> stim.send(ack)
11
>>> cfs = stim.sniff(count=len(fragments) - 1, timeout=1)
>>> for fragment, cf in zip(fragments[1:], cfs):
...     assert (bytes(fragment) == bytes(cf))
... 
>>> thread.join(15)
>>> assert not thread.is_alive()
Traceback (most recent call last):
  File "<input>", line 2, in <module>
AssertionError
UTscapy ended with error code 1
SocketCAN support: False
Using PythonCANSocket virtual
SocketCAN support: False

```

**Comment by polybassa** [comment]
Could you please merge #3378, since even #3379 is failing because of the same reason. 
-- 
Viele Grüße

Nils Weiß

**Comment by polybassa** [comment]
This is on hold until #3487 is merged

**Comment by polybassa** [comment]
Reopen trick
```

###(001)=[failed] Try importing all core separately

>>> import_all(x for x in ALL_FILES if "layers" not in x and "contrib" not in x)
Importing the file 'scapy.sendrecv' failed !
Timed out (>30s)!
Traceback (most recent call last):
  File "<input>", line 2, in <module>
  File "<input>", line 8, in import_all
ImportError


###(002)=[failed] Try importing all layers separately

>>> import_all(x for x in ALL_FILES if "layers" in x)
Importing the file 'scapy.layers.rip' failed !
Timed out (>30s)!
Traceback (most recent call last):
  File "<input>", line 2, in <module>
  File "<input>", line 8, in import_all
ImportError


###(003)=[failed] Try importing all contribs separately

>>> import_all(x for x in ALL_FILES if "contrib" in x)
Importing the file 'scapy.contrib.etherip' failed !
Timed out (>30s)!
Traceback (most recent call last):
  File "<input>", line 2, in <module>
  File "<input>", line 8, in import_all
ImportError
```

```

Regression tests for Scapy
━ Run at 16:00:48 from [test/regression.uts] by UTscapy in 3.9279940128326416
 └ Passed=264
 └ Failed=1

######
## Information on Scapy
######


###(013)=[failed] Configuration conf.use_pcap

>>> if not conf.use_pcap:
...     assert not conf.iface.provider.libpcap
...     conf.use_pcap = True
...     assert conf.iface.provider.libpcap
...     for iface in conf.ifaces.values():
...         assert iface.provider.libpcap or iface.is_valid() == False
...     conf.use_pcap = False
...     assert not conf.iface.provider.libpcap
... 
Traceback (most recent call last):
  File "<input>", line 7, in <module>
AssertionError
```

**Comment by polybassa** [comment]
reopen / close trick. Test failed unrelated to this PR.
``` 
egression tests for Scapy
━ Run at 17:13:36 from [test/regression.uts] by UTscapy in 4.2215211391448975
 └ Passed=264
 └ Failed=1

######
## Information on Scapy
######


###(013)=[failed] Configuration conf.use_pcap

>>> if not conf.use_pcap:
...     assert not conf.iface.provider.libpcap
...     conf.use_pcap = True
...     assert conf.iface.provider.libpcap
...     for iface in conf.ifaces.values():
...         assert iface.provider.libpcap or iface.is_valid() == False
...     conf.use_pcap = False
...     assert not conf.iface.provider.libpcap
... 
Traceback (most recent call last):
  File "<input>", line 7, in <module>
AssertionError
```

**Comment by polybassa** [comment]
@gpotter2  Ready for review. this PR speeds up the unit tests.

**Comment by polydroi** [comment]
@gpotter2 could you please merge this PR. This should speed up the tests quite a lot.

File: test/tools/isotpscanner.uts [hunk 1]
```diff
@@ -98,132 +99,90 @@ assert expected_output in plain_str(std_out)
...
-sniffer.join(timeout=10)
-
-assert 0 == send_returncode
-assert returncode1 == 0 == returncode2
-
-expected_output = [b'tx_id=0x601', b'rx_id=0x700', b'padding=False', b'ext_address=0xbb', b'rx_ext_address=0xaa']
-print(std_out1)
-for out in expected_output:
-    assert plain_str(out) in plain_str(std_out1 + std_out2)
+~ python3_only
```
Comment (gpotter2) [inline test/tools/isotpscanner.uts:156]: I see those but I assume there's a good reason

**Review by gpotter2 — APPROVED** [review]

File: test/tools/isotpscanner.uts (hunk 1 above)
Comment (polydroi) [reply test/tools/isotpscanner.uts:156]: It's because of the mocking.

**Pull request merged**